test/lib/Benchmark.js

```diff
@@ -0,0 +1,182 @@
+/* globals XPCOMUtils, Services, Task */
+"use strict";
+
+const {Cu} = require("chrome");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.import("resource://gre/modules/Services.jsm");
+
+XPCOMUtils.defineLazyModuleGetter(this, "Task",
+                                  "resource://gre/modules/Task.jsm");
+// For the performance.now()
+const performance = Services.appShell.hiddenDOMWindow.performance;
+
+/*
+ * A simple benchmark framework that runs on top of jpm test
```

Comment:
The main use cases we want to benchmark:
* the load latency of A-S with various profile sizes (e.g. history/bookmarks)
* the expensive utility functions or code blocks found by the profiler (e.g. color-analyzer, tippy-top-site, sqlite-queries...)
* new features and its influence on the addon performance